File: src/main/java/nl/helloWorld/helloworld/domain/Counter.java
package nl.helloWorld.helloworld.domain;

public class Counter {
    private int counter;

    Counter(){
        counter = 0;
    }

    public int addNumber(){
        counter += 1;
        return counter;
    }

    public int getCounter() {
        return counter;
    }
}

File: src/main/webapp/counter.js
let addButton = document.getElementById('counterAddButton');
let counter = document.getElementById('counter');

addButton.onclick = function addNumber() {
    const request = new XMLHttpRequest();
    request.open("PATCH", "./counter");
    request.send();

    fetch('./counter')
        .then(response => response.text())
        .then(data => {
            counter.innerHTML = data;
    });
};
